src/liboslexec/shadingsys.cpp

```diff
@@ -3213,7 +3213,8 @@ ShadingSystemImpl::ReParameter(ShaderGroup& group, string_view layername_,
         return false;  // could not find the named layer
 
     // Find the named parameter within the layer
-    int paramindex = layer->findparam(ustring(paramname));
+    int paramindex = layer->findparam(ustring(paramname),
+                                      false /* don't go to master */);
```

Comment:
I have a suggestion:

First, look for it with search_master=false. If that succeeds, we're done.

If that search fails, try again with search_master=true. If that succeeds, then you know it's the case that the parameter did exist, but was optimized away.

I think that in such a case, ReParameter should return `true`. It's not a failure in the same sense as using a parameter that doesn't exist in the shader, or using a parameter not marked as interactive. It's simply that changing the value of the parameter has no effect on the appearance, so you don't need to communicate the new value to the GPU or whatever. But I still consider that a success in terms of changing the value and being able to get the right picture when you rerender.

Reply:
Remember that I'm simultaneously working on another fix in which if you try to ReParameter something that isn't allowed to vary (say, because not folding it would make something illegal for the GPU), the `false` it returns can be interpreted by the renderer as "if I want to vary that parameter, I need to re-send the shader group and re-JIT". So I don't want to throw back `false` failures for harmless things where we can draw the right picture without re-compiling.

Reply:
Makes sense, done.